pypowsybl/perunitnetwork.py

```diff
@@ -0,0 +1,636 @@
+#
+# Copyright (c) 2020, RTE (http://www.rte-france.com)
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+import math as _math
+
+import _pypowsybl
+
+import pypowsybl.network as _net
+import pandas as _pd
+
+
+class PerUnitNetwork(_net.Network):
+    def __init__(self, sn, handle):
+        super().__init__(handle)
+        self.sn = sn
+        self.sqrt3 = _math.sqrt(3)
+
+    def get_buses(self) -> _pd.DataFrame:
+        join = _pd.merge(self.get_elements(_pypowsybl.ElementType.BUS), self.get_voltage_levels()['nominal_v'],
+                         left_on='voltage_level_id', right_index=True)
+        join['v_mag'] = join['v_mag'] / join['nominal_v']
+        return join.drop('nominal_v', axis=1)
+
+    def get_generators(self) -> _pd.DataFrame:
+        """ Get generators as a ``Pandas`` data frame.
+
+        Returns:
+            a generators data frame
+        """
+        generators = self.get_elements(_pypowsybl.ElementType.GENERATOR)
+        generators['target_p'] /= self.sn
+        generators['target_q'] /= self.sn
+        generators['min_p'] /= self.sn
+        generators['min_q'] /= self.sn
+        generators['max_p'] /= self.sn
+        generators['max_q'] /= self.sn
+        generators['p'] /= self.sn
+        generators['q'] /= self.sn
+        join = _pd.merge(generators, self.get_voltage_levels()['nominal_v'],
+                         left_on='voltage_level_id', right_index=True)
+        join['target_v'] = join['target_v'] / join['nominal_v']
+        join['i'] /= self.sn * 10 ** 3 / (self.sqrt3 * join['nominal_v'])
+        return join.drop('nominal_v', axis=1)
+
+    def get_loads(self) -> _pd.DataFrame:
+        """ Get loads as a ``Pandas`` data frame.
+
+        Returns:
+            a loads data frame
+        """
+        loads = self.get_elements(_pypowsybl.ElementType.LOAD)
+        loads['p0'] /= self.sn
+        loads['q0'] /= self.sn
+        loads['p'] /= self.sn
+        loads['q'] /= self.sn
+        join = _pd.merge(loads, self.get_voltage_levels()['nominal_v'],
+                         left_on='voltage_level_id', right_index=True)
+        join['i'] /= self.sn * 10 ** 3 / (self.sqrt3 * join['nominal_v'])
+        return join.drop('nominal_v', axis=1)
+
+    def get_lines(self) -> _pd.DataFrame:
+        """ Get lines as a ``Pandas`` data frame.
+
+        Returns:
+            a lines data frame
+        """
+        lines = self.get_elements(_pypowsybl.ElementType.LINE)
+        lines['p1'] /= self.sn
+        lines['p2'] /= self.sn
+        lines['q1'] /= self.sn
+        lines['q2'] /= self.sn
+        lines = _pd.merge(lines, self.get_voltage_levels()['nominal_v'],
+                          left_on='voltage_level1_id', right_index=True)
```

Comment:
This should be the nominal voltage on side 2 here.

Reply:
on a line nominal voltage is the same on both sides isn't it ?